src/fontcache.cpp

```diff
@@ -1056,9 +1056,301 @@ static void LoadWin32Font(FontSize fs)
+			if (data) {
+				uint16_t lowestRecPPEM; // At offset 46 of the 'head' TrueType table.
+				CFDataGetBytes(data.get(), CFRangeMake(46, sizeof(lowestRecPPEM)), (UInt8 *)&lowestRecPPEM);
+				min_size = CFSwapInt16BigToHost(lowestRecPPEM); // TrueType data is always big-endian.
+			} else {
+				CFAutoRelease<CFNumberRef> size((CFNumberRef)CTFontCopyAttribute(f.get(), kCTFontSizeAttribute));
+				CFNumberGetValue(size.get(), kCFNumberFloatType, &min_size);
+			}
+
+			/* Font height is minimum height plus the difference between the default
+			 * height for this font size and the small size. */
+			int diff = scaled_height - ScaleFontTrad(_default_font_height[FS_SMALL]);
+			pixels = Clamp(std::min(min_size, 20.0f) + diff, scaled_height, MAX_FONT_SIZE);
+		}
+	} else {
+		pixels = ScaleFontTrad(pixels);
+	}
+	this->used_size = pixels;
+
+	this->font.reset(CTFontCreateWithFontDescriptor(this->font_desc.get(), pixels, nullptr));
+
+	/* Query the font metrics we needed. We generally round all values up to
+	 * make sure we don't inadvertently cut off a row or column of pixels.*/
+	this->units_per_em = CTFontGetUnitsPerEm(this->font.get());
+	this->ascender = (int)std::ceil(CTFontGetAscent(this->font.get()));
+	this->descender = -(int)std::ceil(CTFontGetDescent(this->font.get()));
+	this->height = this->ascender - this->descender;
+
+	/* Get real font name. */
+	char name[128];
+	CFAutoRelease<CFStringRef> font_name((CFStringRef)CTFontCopyAttribute(this->font.get(), kCTFontDisplayNameAttribute));
+	CFStringGetCString(font_name.get(), name, lengthof(name), kCFStringEncodingUTF8);
+	this->font_name = name;
+
+	DEBUG(freetype, 2, "Loaded font '%s' with size %d", this->font_name.c_str(), pixels);
+}
+
+GlyphID CoreTextFontCache::MapCharToGlyph(WChar key)
+{
+	assert(IsPrintable(key));
+
+	if (key >= SCC_SPRITE_START && key <= SCC_SPRITE_END) {
+		return this->parent->MapCharToGlyph(key);
+	}
+
+	/* Convert characters outside of the BMP into surrogate pairs. */
+	UniChar chars[2];
+	if (key >= 0x010000U) {
+		chars[0] = (UniChar)(((key - 0x010000U) >> 10) + 0xD800);
+		chars[1] = (UniChar)(((key - 0x010000U) & 0x3FF) + 0xDC00);
+	} else {
+		chars[0] = (UniChar)(key & 0xFFFF);
+	}
+
+	CGGlyph glyph[2] = {0, 0};
+	if (CTFontGetGlyphsForCharacters(this->font.get(), chars, glyph, key >= 0x010000U ? 2 : 1)) {
+		return glyph[0];
+	} else {
+		return 0;
+	}
+}
+
+const void *CoreTextFontCache::InternalGetFontTable(uint32 tag, size_t &length)
+{
+	CFAutoRelease<CFDataRef> data(CTFontCopyTable(this->font.get(), (CTFontTableTag)tag, kCTFontTableOptionNoOptions));
+	if (!data) return nullptr;
+
+	length = CFDataGetLength(data.get());
+	auto buf = MallocT<UInt8>(length);
+
+	CFDataGetBytes(data.get(), CFRangeMake(0, (CFIndex)length), buf);
+	return buf;
+}
+
+const Sprite *CoreTextFontCache::InternalGetGlyph(GlyphID key, bool aa)
+{
+	/* Get glyph size. */
+	CGGlyph glyph = (CGGlyph)key;
+	CGRect bounds = CGRectNull;
+	if (MacOSVersionIsAtLeast(10, 8, 0)) {
```

Comment:
Do we support 10.8? Weren't we 10.9+? (our description of the downloads suggests we are)